extensions/amp-analytics/0.1/transport.js

```diff
@@ -118,13 +124,143 @@ export class Transport {
   }
+
+  /**
+   * Sends an Amp Analytics trigger event to a vendor's cross-domain iframe,
+   * or queues the message if the frame is not yet ready to receive messages.
+   * TODO: Implement throttling if messages are sent too rapidly.
+   * @param {string} request
+   * @param {!Object<string, string>} transportOptions
+   */
+  static sendRequestUsingCrossDomainIframe(request, transportOptions) {
+    const frameUrl = Transport.appendHashToUrl_(transportOptions['iframe'],
+      transportOptions['dataHash']);
+    const frameData = Transport.crossDomainFrames[frameUrl];
+    if (frameData.isReady) {
+      this.sendToCrossDomainIframe_(frameData.frame, [request]);
+    } else {
+      frameData.msgQueue.push(request);
+    }
+  }
+
+  /**
+   * Takes a URL and a data hash (the part that comes after the '#',
+   * optionally including the '#' itself) and concatenates them, adding the
+   * '#' if necessary
+   * @param {!string} url
+   * @param {string=} dataHash
+   * @return {!string}
+   * @private
+   */
+  static appendHashToUrl_(url, dataHash) {
+    if (dataHash) {
+      return url + (startsWith(dataHash, '#') ? dataHash : '#' + dataHash);
+    }
+    return url;
+  }
+
+  /**
+   * If iframe (and optionally dataHash as well) are specified in
+   * config/transport, check whether third-party iframe already exists, and if
+   * not, create it.
+   * @param {!HTMLDocument} ampDoc The AMP document
+   * @param {!Object<string,string>} transportOptions The 'transport' portion
+   * of the amp-analytics
+   * config object
+   */
+  static processCrossDomainIframe(ampDoc, transportOptions) {
+    user().assert(!(transportOptions['beacon'] || transportOptions['xhrpost'] ||
+      transportOptions['image']), 'Cross-domain frame cannot coexist with' +
+      ' other transport methods');
+    const frameUrl = Transport.appendHashToUrl_(transportOptions['iframe'],
+        transportOptions['dataHash']);
+    // If iframe doesn't exist for this iframe url (and data hash), create it.
+    if (!Transport.hasCrossDomainFrame(frameUrl)) {
+      const frame = Transport.createCrossDomainFrame(ampDoc, frameUrl);
+      ampDoc.body.appendChild(frame);
+    }
+  }
+
+  /**
+   * Returns whether a url of a cross-domain frame is already known
+   * @param {!string} frameUrl
+   * @return {!boolean}
+   */
+  static hasCrossDomainFrame(frameUrl) {
+    return hasOwn(Transport.crossDomainFrames, frameUrl);
+  }
+
+  /**
+   * Create a cross-domain iframe for third-party vendor anaytlics
+   * @param {!HTMLDocument} ampDoc  The document node of the parent page
+   * @param {!string} frameUrl  The URL, including data hash if
+   * applicable, of the cross-domain iframe
+   * @return {!Element}
+   */
+  static createCrossDomainFrame(ampDoc, frameUrl) {
+    // DO NOT MERGE THIS
+    // Warning: the scriptSrc URL below is only temporary. Don't merge
+    // before getting resolution on that.
+    const frame = createElementWithAttributes(ampDoc, 'iframe', {
```

Comment:
Pointing out this comment again since previous one got marked outdated. In previous thread, I suggested discussing what the actual value of scriptSrc should be.